### ISSUE TITLE
response.text() and response.json() get "stuck" sometimes

### ISSUE THREAD

**Issue by mikelambert** [facebook/react-native#14391]
My code more or less does the following, on React Native 0.45:
```
async function request() {
    const result = await fetch(url);
    const data = await result.text();
    console.log(data);
    return data;
}
```

When running this in Chrome Debugger on the iOS Simulator, the result will get "stuck" sometimes. By that, I mean the first time I call it, it works fine. But the second time I run a request, the `data` will not be printed out until something else happens to jiggle the event loop and cause it to return.

If I disable the chrome debugger, things work.

I assume this is related to what people have been complaining about on the prematurely-closed bug https://github.com/facebook/react-native/issues/6679 .

=================

What follows is my attempt at debugging. I won't promise anything, as I'm not a promise expert, but maybe it helps narrow things down?

The http response data is shuffled into the `response.blob`. When `text()` or `json()` is called, it calls `readBlobAsText` to read the blob, and I do see the string successfully loaded and passed around within the `promise/setimmediate/core.js` code. Unfortunately, this string doesn't always make it back to the caller for some reason.

(This might be a red herring, but I see it call `handleResolved()` and the equivalent of `setImmediate((self, deferred) => tryCallOne(deferred.onFulfilled, self._65 /* body payload */))`, but the anonymous function isn't actually called immediately...)

And interesting to note, if I edit `fetch.js` to disable `support.blob`, it also works.

This makes me suspect it has something to do with the `text()` function. In most cases, it returns `Promise.reject` or `Promise.resolve`, but in the case of using blobs, it returns a newly-generated `Promise` (from the `fileReaderReady` function). My guess is that this probably triggers another level of promise-indirection, that somehow causes it to stall before the event-loop notices it and triggers it to complete.

I'm still quite a newb as to the `JSTimers*` code and guts of the `promise` code, so this is where the trail goes cold for me. :(

**Comment by wschurman** [comment]
Seeing this as well. Commenting to bump.

**Comment by congqiao** [comment]
Same here. It seems this only happens in React Native 0.45. For me the probability is around 50%.

Quick workaround: Tap the screen to continue.

**Comment by kyo504** [comment]
I also experienced this behavior after upgrading to 0.45.

**Comment by EdwardHuCS** [comment]
Just upgraded and ran into this bug. Using @sebandres' quick hack to force it to complete.
```
fetch(url).then(response => {
  setTimeout(() => null, 0);  // workaround for issue-6679
  return response.json();
})
```

**Comment by mikelambert** [comment]
And the modified ES7 form of the above code that also works for me (thanks!):
```
    setTimeout(() => null, 0);
    const json = await result.json();
```

Just add the setTimeout before you call `json()` or `text()` or anything else that operates on `response.blob`.

As far as the underlying bug, definitely feels like the eventloop "falls asleep" when processing `response.blob` and `readBlobAsText`, instead of recognizing there are still a `Promise` on the eventloop to process and return data back to the callers.

**Comment by mikelambert** [comment]
And I just found this comment (in a "closed" bug), which might describe the root cause of this issue: https://github.com/facebook/react-native/issues/9436#issuecomment-302868758

**Comment by mikelambert** [comment]
Yay, this appears to be fixed in 0.47 by https://github.com/facebook/react-native/commit/64825389dfb4e01d5b30cd63b0b41937a9bb431d !

**Comment by Arshiamidos** [comment]
thanks @EdwardHuCS
